src/globals.c

```diff
@@ -60,51 +60,156 @@ bool download_only;
 bool have_network = false;	    /* assume no access until proved */
-#define URL_COUNT 2
-char *version_server_urls[URL_COUNT] = {
-	NULL,
-	"https://download.clearlinux.org/update",
-};
-char *content_server_urls[URL_COUNT] = {
-	NULL,
-	"https://download.clearlinux.org/update",
-};
-char *preferred_version_url;
-char *preferred_content_url;
+char *version_url = NULL;
+char *content_url = NULL;
 long update_server_port = -1;
 
-#define SWUPD_DEFAULT_FORMAT "3"
-bool set_format_string(char *userinput)
+static const char *default_version_url_path = "/usr/share/defaults/swupd/versionurl";
+static const char *default_content_url_path = "/usr/share/defaults/swupd/contenturl";
+static const char *default_format_path = "/usr/share/defaults/swupd/format";
+
+static int set_default_value(char **global, const char *path)
 {
-	int version;
+	char line[LINE_MAX];
+	FILE *file;
+	char *c;
+	char *rel_path;
+	int ret = -1;
 
-	if (userinput == NULL) {
-		if (format_string) {
-			free(format_string);
+	string_or_die(&rel_path, "%s%s", path_prefix, path);
+
+	file = fopen(rel_path, "r");
+	if (!file) {
+		printf("Error: Unable to open %s\n", rel_path);
+		free(rel_path);
+		return ret;
+	}
+
+	/* the file should contain exactly one line */
+	line[0] = 0;
+	if (fgets(line, LINE_MAX, file) == NULL) {
+		if (ferror(file)) {
+			printf("Error: Unable to read data from %s\n", rel_path);
+			goto fail;
+		}
+		if (feof(file)) {
+			printf("Error: Contents of %s are empty\n", rel_path);
+			goto fail;
 		}
-		string_or_die(&format_string, "%s", SWUPD_DEFAULT_FORMAT);
-		return true;
 	}
 
-	// allow "staging" as a format string
-	if ((strcmp(userinput, "staging") == 0)) {
-		if (format_string) {
-			free(format_string);
+	/* remove newline if present */
+	c = strchr(line, '\n');
+	if (c) {
+		*c = '\0';
+	}
+
+	string_or_die(global, "%s", line);
+	ret = 0;
+fail:
+	fclose(file);
+	free(rel_path);
+	return ret;
+
+}
+
+static int set_url(char **global, char *url, const char *path)
+{
+	int ret = 0;
+
+	if (url) {
+		if (*global) {
+			free(*global);
 		}
-		string_or_die(&format_string, "%s", userinput);
-		return true;
+		string_or_die(global, "%s", url);
+	} else {
+		if (*global) {
+			/* option passed on command line previously */
+			return ret;
+		} else {
+			/* no option passed; use the default value */
+			ret = set_default_value(global, path);
+			if (ret < 0) {
+				return ret;
+			}
+		}
+	}
+
+	return ret;
+}
+
+void set_content_url(char *url) {
+	int ret;
+
+	ret = set_url(&content_url, url, default_content_url_path);
+	if (ret < 0) {
+		printf("\nDefault content URL cannot be read. Use the -c option instead.\n");
+		exit(EXIT_FAILURE);
 	}
+}
 
-	// otherwise, expect a positive integer
+void set_version_url(char *url) {
+	int ret;
+
+	ret = set_url(&version_url, url, default_version_url_path);
+	if (ret < 0) {
+		printf("\nDefault version URL cannot be read. Use the -v option instead.\n");
+		exit(EXIT_FAILURE);
+	}
+}
```

Comment:
Print them out in general? I'm not certain the end user would be interested in this info, since it is more of an internal detail. Maybe we could add a --print-config option?